### PR TITLE
scriptix: fix broken build

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

The port is currently broken across the board: https://ports.macports.org/port/scriptix/details/
Here is why it fails: https://trac.macports.org/ticket/65727
Solution: https://github.com/Papierkorb/bindgen/issues/77
More info: https://github.com/ivmai/bdwgc/issues/268

Second commit adds modeline, fixes lint warnings, adds proper checksums, changes dead homepage to sourceforce repo.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 Server
Xcode 3.2.6

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by barracuda156** [comment]
Okay, `clang` builds need extra fixes. Now checks fail on several errors of this kind:
```
error: no matching constructor for initialization of 'Scriptix::Compiler::CompilerNode'
```

**Comment by barracuda156** [comment]
@ryandesign Let me know if we just remove the port. I will close this PR then.

**Comment by mascguy** [comment]
> Okay, `clang` builds need extra fixes. Now checks fail on several errors of this kind:
> 
> ```
> error: no matching constructor for initialization of 'Scriptix::Compiler::CompilerNode'
> ```

Sergey, now that you've got much more experience under your belt, would you be comfortable with trying to fix this?

**Comment by barracuda156** [comment]
@mascguy I can try, but perhaps no point in wasting too much of the effort here, given what @ryandesign wrote: https://trac.macports.org/ticket/65727#comment:1

**Comment by barracuda156** [comment]
(Let me rebase to get new logs.)

**Comment by barracuda156** [comment]
@mascguy So it is not a Clang issue as such, but rather an issue of archaic code. I just tried to build it with `gcc12`, and it fails with similar errors:
```
In file included from grammar.yy:37:
../include/compiler.h:207:27: error: conflicting declaration 'Scriptix::TypeInfo* type'
  207 |                 TypeInfo* type,
      |                 ~~~~~~~~~~^~~~
../include/compiler.h:201:21: note: previous declaration as 'int type'
  201 |                 int type,
      |                 ~~~~^~~~
grammar.cc: In function 'size_t yystrlen(const char*)':
grammar.cc:1032:24: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1032 |   register const char *yys = yystr;
      |                        ^~~
grammar.cc: In function 'char* yystpcpy(char*, const char*)':
grammar.cc:1057:18: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1057 |   register char *yyd = yydest;
      |                  ^~~
grammar.cc:1058:24: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1058 |   register const char *yys = yysrc;
      |                        ^~~
grammar.cc: In function 'int yyparse()':
grammar.cc:1193:16: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1193 |   register int yystate;
      |                ^~~~~~~
grammar.cc:1194:16: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1194 |   register int yyn;
      |                ^~~
grammar.cc:1212:23: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1212 |   register short int *yyssp;
      |                       ^~~~~
grammar.cc:1217:21: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1217 |   register YYSTYPE *yyvsp;
      |                     ^~~~~
In file included from lexer.ll:36:
../include/compiler.h:207:27: error: conflicting declaration 'Scriptix::TypeInfo* type'
  207 |                 TypeInfo* type,
      |                 ~~~~~~~~~~^~~~
../include/compiler.h:201:21: note: previous declaration as 'int type'
  201 |                 int type,
      |                 ~~~~^~~~
lexer.cc: In function 'int yylex()':
lexer.cc:700:32: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  700 |         register yy_state_type yy_current_state;
      |                                ^~~~~~~~~~~~~~~~
lexer.cc:701:24: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  701 |         register char *yy_cp, *yy_bp;
      |                        ^~~~~
lexer.cc:701:32: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  701 |         register char *yy_cp, *yy_bp;
      |                                ^~~~~
lexer.cc:702:22: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  702 |         register int yy_act;
      |                      ^~~~~~
lexer.cc:751:42: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  751 |                         register YY_CHAR yy_c = yy_ec[YY_SC_TO_UI(*yy_cp)];
      |                                          ^~~~
lexer.cc: In function 'int yy_get_next_buffer()':
lexer.cc:1154:24: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1154 |         register char *dest = YY_CURRENT_BUFFER_LVALUE->yy_ch_buf;
      |                        ^~~~
lexer.cc:1155:24: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1155 |         register char *source = (yytext_ptr);
      |                        ^~~~~~
lexer.cc:1156:22: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1156 |         register int number_to_move, i;
      |                      ^~~~~~~~~~~~~~
lexer.cc:1156:38: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1156 |         register int number_to_move, i;
      |                                      ^
lexer.cc: In function 'yy_state_type yy_get_previous_state()':
lexer.cc:1280:32: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1280 |         register yy_state_type yy_current_state;
      |                                ^~~~~~~~~~~~~~~~
lexer.cc:1281:24: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1281 |         register char *yy_cp;
      |                        ^~~~~
lexer.cc:1287:34: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1287 |                 register YY_CHAR yy_c = (*yy_cp ? yy_ec[YY_SC_TO_UI(*yy_cp)] : 1);
      |                                  ^~~~
lexer.cc: In function 'yy_state_type yy_try_NUL_trans(yy_state_type)':
lexer.cc:1312:22: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1312 |         register int yy_is_jam;
      |                      ^~~~~~~~~
lexer.cc:1313:24: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1313 |         register char *yy_cp = (yy_c_buf_p);
      |                        ^~~~~
lexer.cc:1315:26: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1315 |         register YY_CHAR yy_c = 1;
      |                          ^~~~
lexer.cc: At global scope:
lexer.cc:1333:49: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1333 |     static void yyunput (int c, register char * yy_bp )
      |                                                 ^~~~~
lexer.cc: In function 'void yyunput(int, char*)':
lexer.cc:1335:24: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1335 |         register char *yy_cp;
      |                        ^~~~~
lexer.cc:1345:30: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1345 |                 register int number_to_move = (yy_n_chars) + 2;
      |                              ^~~~~~~~~~~~~~
lexer.cc:1346:32: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1346 |                 register char *dest = &YY_CURRENT_BUFFER_LVALUE->yy_ch_buf[
      |                                ^~~~
lexer.cc:1348:32: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
 1348 |                 register char *source =
      |                                ^~~~~~
 /opt/local/bin/g++-mp-12 -DHAVE_CONFIG_H -I. -I. -I../include -I../include -I/opt/local/include -I/opt/local/include -pipe -Os -arch ppc -MT parser.lo -MD -MP -MF .deps/parser.Tpo -c parser.cc  -fno-common -DPIC -o .libs/parser.o
make[1]: *** [lexer.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [grammar.lo] Error 1
 /opt/local/bin/g++-mp-12 -DHAVE_CONFIG_H -I. -I. -I../include -I../include -I/opt/local/include -I/opt/local/include -pipe -Os -arch ppc -MT value.lo -MD -MP -MF .deps/value.Tpo -c value.cc -o value.o >/dev/null 2>&1
 /opt/local/bin/g++-mp-12 -DHAVE_CONFIG_H -I. -I. -I../include -I../include -I/opt/local/include -I/opt/local/include -pipe -Os -arch ppc -MT system.lo -MD -MP -MF .deps/system.Tpo -c system.cc -o system.o >/dev/null 2>&1
 /opt/local/bin/g++-mp-12 -DHAVE_CONFIG_H -I. -I. -I../include -I../include -I/opt/local/include -I/opt/local/include -pipe -Os -arch ppc -MT type.lo -MD -MP -MF .deps/type.Tpo -c type.cc -o type.o >/dev/null 2>&1
In file included from parser.cc:36:
../include/compiler.h:207:27: error: conflicting declaration 'Scriptix::TypeInfo* type'
  207 |                 TypeInfo* type,
      |                 ~~~~~~~~~~^~~~
../include/compiler.h:201:21: note: previous declaration as 'int type'
  201 |                 int type,
      |                 ~~~~^~~~
parser.cc:292:1: error: no declaration matches 'Scriptix::Compiler::CompilerNode::CompilerNode(Scriptix::Compiler::Compiler*, int, Scriptix::Compiler::CompilerNode*, Scriptix::Compiler::CompilerNode*, Scriptix::Compiler::CompilerNode*, Scriptix::Compiler::CompilerNode*, Scriptix::NameID, Scriptix::TypeInfo*, Scriptix::Value*, int)'
  292 | Scriptix::Compiler::CompilerNode::CompilerNode(Scriptix::Compiler::Compiler* s_info,
      | ^~~~~~~~
../include/compiler.h:184:8: note: candidates are: 'constexpr Scriptix::Compiler::CompilerNode::CompilerNode(Scriptix::Compiler::CompilerNode&&)'
  184 | struct CompilerNode : public GC::Collectable {
      |        ^~~~~~~~~~~~
../include/compiler.h:184:8: note:                 'constexpr Scriptix::Compiler::CompilerNode::CompilerNode(const Scriptix::Compiler::CompilerNode&)'
../include/compiler.h:200:9: note:                 'Scriptix::Compiler::CompilerNode::CompilerNode(...)'
  200 |         CompilerNode(Compiler* info,
      |         ^~~~~~~~~~~~
../include/compiler.h:184:8: note: 'struct Scriptix::Compiler::CompilerNode' defined here
  184 | struct CompilerNode : public GC::Collectable {
      |        ^~~~~~~~~~~~
make[1]: *** [parser.lo] Error 1
make[1]: Leaving directory `/opt/local/var/macports/build/_opt_PPCRosettaPorts_devel_scriptix/scriptix/work/scriptix-0.31/lib'
make: *** [all-recursive] Error 1
```
I am not too inclined to struggle with fixing this, TBH.

Two easy options are:
1. Close PR, remove the port from Macports.
2. Set it to be build with `apple-gcc42` on whatever systems support that compiler, like it was done for some very old Haskell compilers.

What do you think?

**Comment by mascguy** [comment]
> @mascguy So it is not a Clang issue as such, but rather an issue of archaic code. I just tried to build it with `gcc12`, and it fails with similar errors:
> [...]
> I am not too inclined to struggle with fixing this, TBH.
> 
> Two easy options are:
>     1. Close PR, remove the port from Macports.
>     2. Set it to be build with `apple-gcc42` on whatever systems support that compiler, like it was done for some very old Haskell compilers.
> 
> What do you think?

Well, port `awemud` depends on `scriptix`, so that would have to be removed too.

So I'm inclined to suggest option 2. (And it's probably fixable with patches, if folks want to do so. But let's forget that for now.)

**Comment by barracuda156** [comment]
@mascguy Done, but it cannot be tested on CI, so kinda a blind shot.

File: devel/scriptix/Portfile [hunk 1]
```diff
@@ -1,23 +1,52 @@
...
+    }
+
+    if {${os.major} > 15} {
+        known_fail  yes
+        pre-fetch {
+            ui_error "${name} requires MacOS 10.11 or earlier"
+            return -code error "incompatible OS version"
+        }
+    }
+}
```
Comment (mascguy) [inline devel/scriptix/Portfile:46]: While it makes sense to limit this to 10.11 or earlier, which compiler will be used for 10.4 thru 10.6?

Comment (barracuda156) [reply devel/scriptix/Portfile:46]: Xcode gcc. Should we specify it explicitly? I though it should be just picked.

Comment (mascguy) [reply devel/scriptix/Portfile:46]: > Xcode gcc. Should we specify it explicitly? I thought it should be just picked.

Yep, just tested on 10.5, and builds fine. So we're good-to-go, no other changes necessary.

**Review by mascguy — APPROVED** [review]
Tested locally on 10.5, and looks good. Merging.

File: devel/scriptix/Portfile (hunk 1 above)
Comment (jmroot) [inline devel/scriptix/Portfile:46]: This should be written as
```
platforms {darwin < 16}
```

**Pull request merged**